### PR TITLE
Use new story format in official storybook

### PR DESCRIPTION
**Pull request by tmeasday**
Moving #4885 over here -- for #4848.

As we work on the new UI, let's write our stories in the new format.

@ndelangen would you be interested in pairing on improving this `importAll` function? How's your HMR? This works as is but makes use of `window` and is a bit of a hack.

### CODE REVIEW

File: examples/official-storybook/config.js [hunk 1]
```diff
@@ -60,8 +62,60 @@ addParameters({
   },
 });
 
-function importAll(req) {
-  req.keys().forEach(filename => req(filename));
+// TODO -- somehow don't use window for this cross HMR storage
+const { previousExports = {} } = window; // eslint-disable-line no-undef
```
Comment (ndelangen) [inline examples/official-storybook/config.js]: You can use module.hot.data:

https://webpack.js.org/api/hot-module-replacement/#dispose-or-adddisposehandler-

```
dispose (or addDisposeHandler) 
Add a handler which is executed when the current module code is replaced. This should be used to remove any persistent resource you have claimed or created. If you want to transfer state to the updated module, add it to given data parameter. This object will be available at module.hot.data after the update.

module.hot.dispose(data => {
  // Clean up and pass data to the updated module...
});
```

Comment (tmeasday) [reply examples/official-storybook/config.js]: OK, that worked, thanks @ndelangen

**Comment by tmeasday** [comment]
So I think this is OK to merge. In terms of a user API, I think we should either export a function:

```js
importAll(require.context(...), module);
```

or modify the configure API to also accept an array of contexts:

```js
configure([ require.context(...), require.context(...)], module);
```

What do you think @ndelangen?

**Review by ndelangen — APPROVED** [review]

File: examples/official-storybook/stories/core/parameters.stories.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+export default {
+  title: 'Core|Parameters',
+  decorators: [fn => fn({ parameters: { decoratorParameter: 'decoratorParameter' } })],
+  parameters: {
+    chapterParameter: 'chapterParameter',
+  },
+};
+
+// I'm not sure what we should recommend regarding propTypes? are they a good idea for examples?
+// Given we sort of control the props, should we export a prop type?
```
Comment (Hypnosphi) [inline examples/official-storybook/stories/core/parameters.stories.js]: Technically, story itself is not a React component: decorators call it as a function, and propTypes have no effect in that case

Comment (ndelangen) [reply examples/official-storybook/stories/core/parameters.stories.js]: In react they are eventually invoked as a React Component in the renderer though?

Comment (Hypnosphi) [reply examples/official-storybook/stories/core/parameters.stories.js]: Not the inner story function but `decoratedStory`

**Pull request merged**